pkgs/by-name/pr/protonmail-bridge-gui/package.nix

```diff
@@ -60,6 +60,18 @@ stdenv.mkDerivation (finalAttrs: {
 
     # Don't build `bridge-gui-tester`
     sed -i "/add_subdirectory(bridge-gui-tester)/d" CMakeLists.txt
+
+    # Qt 6.8 support
+    # fix "ColorImage is not a type"
+    find . -name '*.qml' -type f -exec sed -i '
+    /import QtQuick.Controls/ {
+      n
+      /import QtQuick.Controls.impl/! {
+        i import QtQuick.Controls.impl
+      }
+    }' {} +
+    # fix "Cannot override FINAL property"
+    find . -name '*.qml' -type f -exec sed -i 's/\bpopupType\b/protonPopupType/g' {} +
```

Comment:
Wouldn't it be easier to maintain a patch?

Reply:
I thought it would be "easier" (less work for me after each upstream update) to maintain a `sed` script than a big patch affecting dozens of files (that I would generate after running these commands on the new upstream source anyway).

I want to follow best practices though and would be happy to defer to your experience in the matter! I came across this [thread in the forum](https://discourse.nixos.org/t/when-to-use-substituteinplace-functions-vs-a-patch/11073/12) and this [blog post](https://drakerossman.com/blog/how-to-patch-a-package-source-on-nixos) that suggest it's better to use [`substituteInPlace`](https://nixos.org/manual/nixpkgs/unstable/#fun-substituteInPlace) instead of `find...-exec...sed -i...`, and Mic92's observation that
> A patch file usually has the advantage that it fails loudly when it no longer applies.

Let me have a think about the best way forward. If you have any other links re: best practices for this I'll read and consider them, as what I decide to do should be consistently applied across the derivation's entire `postPatch` section.

Reply:
The sed is so complex that when it fails I couldn't fix it. A patch could be applied on the source tree and more easily be fixed there.

Reply:
I'll remember to KISS!